awx/ui/src/screens/WorkflowApproval/WorkflowApprovalDetail/WorkflowApprovalDetail.js

```diff
@@ -169,31 +193,19 @@ function WorkflowApprovalDetail({ workflowApproval }) {
         />
       </DetailList>
       <CardActionsRow>
-        {workflowApproval.can_approve_or_deny && (
-          <>
-            <Button
-              ouiaId={`${workflowApproval.id}-approve-button`}
-              aria-label={t`Approve`}
-              variant="primary"
-              onClick={approveWorkflowApproval}
-              isDisabled={isLoading}
-            >
-              {t`Approve`}
-            </Button>
-            <Button
-              ouiaId={`${workflowApproval.id}-deny-button`}
-              aria-label={t`Deny`}
-              variant="danger"
-              onClick={denyWorkflowApproval}
-              isDisabled={isLoading}
-            >
-              {t`Deny`}
-            </Button>
-          </>
-        )}
+        {workflowApproval.status === 'pending' &&
+          workflowApproval.can_approve_or_deny && (
+            <WorkflowApprovalControls
+              selected={[workflowApproval]}
+              onApprove={approveWorkflowApproval}
+              onHandleDeny={denyWorkflowApproval}
+              onHandleCancel={handleCancel}
+              onToggleToolbarKebab={(isOpen) => setIsKebabModalOpen(isOpen)}
+              isKebabOpen={isKebabOpen}
+            />
+          )}
         {workflowApproval.status !== 'pending' &&
-          workflowApproval.summary_fields.user_capabilities &&
-          workflowApproval.summary_fields.user_capabilities.delete && (
+          workflowApproval.summary_fields?.user_capabilities?.delete && (
```

Comment:
I got an error when attempt to delete the work flow briefly after was cancelled.

<img width="1502" alt="image" src="https://user-images.githubusercontent.com/9053044/160491380-8382191c-142f-4757-8471-1978f0926dc3.png">


If I wait a bit longer to delete after the work flow job is cancelled. No error is raised.

<img width="1790" alt="image" src="https://user-images.githubusercontent.com/9053044/160491558-2fd59fc7-c4d5-436b-9c41-65f2649b0f74.png">